src/databricks/labs/dqx/engine.py

```diff
@@ -384,16 +395,19 @@ def _create_results_map(self, df: DataFrame, checks: list[DQRule], dest_col: str
 
         check_cols = []
         for check in checks:
+            user_metadata = self.user_metadata
+            if check.user_metadata is not None:
+                user_metadata |= check.user_metadata
```

Comment:
Add a comment pls that if a the same user metadata exists is already defined at the engine level it will get overwritten by the user metadata from the check.
Please add this info the documentation as well.